### PR TITLE
Handle the app flow based on user role in app

### LINKED ISSUE #84: Add model and viewmodel layer for WoofApp to handle app flow
## Depends on tasks:

- #79
- #110 

## Description
- [x] create model layer containing information about user preferences `Preferences` - #111 
- [ ] create ViewModel that will provide data or status to handle AppFlow (sitter or owner side).

#### `WoofApp.swift`
- create model named `Preferences` containing next fields
```swift
var selectedRole: Role?
```

```swift
enum Role{
    case owner, sitter
}
```
- creating a ViewModel in which the following should be implemented
- creating property to inform view(WoofApp) about loaded selected role
``` swift
@Published var loadRole: Role? 
```
- creating a read method to get the selected role from preferences KeyValueStorage
- creating tests for methods of ViewModel, if needed.

### LINKED ISSUE #85: Creating logical flow to display proper view based on the selected role
## Depends on tasks:
- #79
- #84

## Description
Creating logical flow to display proper view based on the selected role

### Dev requirements and restrictions:
- edit the WoofApp file to have a flow that, based on the role, will show one of the three views
- initialize WoofAppViewModel in `WoofApp.swift` and check the current user role and display the proper view.

### PR DESCRIPTION
**Pull request by NadzeyaShpakouskaya**
## Summary
This PR is a part of the work to resolve #84 and resolve #85.
Display different views based on the user role that was stored in user preferences.

### NOTE!
To test that expected view will be presented you can simulate different user roles in`init()` method in `StartScreenViewModel` instead of `userRole = PreferencesHandler.getUserRole()`

```swift 
init() {
   // userRole = PreferencesHandler.getUserRole()
   userRole = .owner
   // userRole = .sitter
   // userRole = .none
}
```

## Changes
- create ViewModel that provides a user role to handle AppFlow (sitter or owner side).
- update the window group in `WoofApp` to display the view depending on the selected role.

## Checks to complete
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] PR's title has a conventional commit style (short and descriptive).
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/ViewModel/StartScreenViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+/// Responsible for managing the data and behavior of the start screen view.
+final class StartScreenViewModel: ObservableObject {
+    /// The user role that defines the app flow, it's `none` by default.
+    @Published var userRole: Role = .none
```
Comment (marinkman) [inline Woof/Woof/ViewModel/StartScreenViewModel.swift:6]: And why assign the property the role of .none, if .none is assigned during initialization, if the role is not selected?

**Review by marinkman — COMMENTED** [review]
Good job. I have a question.

File: Woof/Woof/ViewModel/StartScreenViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+    init() {
```
Comment (YuliaNastiushkina) [inline Woof/Woof/ViewModel/StartScreenViewModel.swift:8]: Missed documentation.

**Review by YuliaNastiushkina — CHANGES_REQUESTED** [review]
Good job!
Just one moment, check it please.

**Review by kirpichonok — APPROVED** [review]
LGTM, but you should resolve the conflict.

File: Woof/Woof/ViewModel/StartScreenViewModel.swift (hunk 1 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/ViewModel/StartScreenViewModel.swift]: Fixed with https://github.com/ios-course/ironfoudation-team-project/pull/156/commits/2291d80754fc4ddd7c76e05f5b0e21f133902573

File: Woof/Woof/ViewModel/StartScreenViewModel.swift (hunk 2 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/ViewModel/StartScreenViewModel.swift:9]: Fixed with https://github.com/ios-course/ironfoudation-team-project/pull/156/commits/2291d80754fc4ddd7c76e05f5b0e21f133902573

**Review by YuliaNastiushkina — APPROVED** [review]
LGTM

**Pull request merged**